app/editor/src/features/admin/users/UserList.tsx

```diff
@@ -11,30 +14,62 @@ import { UserFilter } from './UserFilter';
 
 export const UserList: React.FC = () => {
   const navigate = useNavigate();
-  const [{ users }, api] = useUsers();
-  const [{ requests }] = useApp();
-
+  const [{ users, userFilter }, { findUsers, storeFilter }] = useUsers();
+  const [{ requests }, { isUserReady }] = useApp();
   const [page, setPage] = React.useState(
     new Page(users.page - 1, users.quantity, users.items, users.total),
   );
 
+  const handleChangeSort = React.useCallback(
+    (sortBy: SortingRule<IUserModel>[]) => {
+      const sorts = sortBy.map((sb) => ({ id: sb.id, desc: sb.desc }));
+      const same = sorts.every(
+        (val, i) => val.id === userFilter.sort[i]?.id && val.desc === userFilter.sort[i]?.desc,
+      );
+      if (!same) {
+        storeFilter({ ...userFilter, sort: sorts });
+      }
+    },
+    [storeFilter, userFilter],
+  );
+
+  const fetch = React.useCallback(
+    async (filter: IUserFilter) => {
+      try {
+        const data = await findUsers(makeUserFilter(filter));
+        const page = new Page(data.page - 1, data.quantity, data?.items, data.total);
+
+        setPage(page);
+        // return page;
+      } catch (error) {
+        // TODO: Handle error
+        throw error;
+      }
+    },
+    [findUsers],
+  );
+
   React.useEffect(() => {
-    if (!users.items.length) {
-      api.findUsers({ isSystemAccount: false }).then((data) => {
-        setPage(new Page(data.page - 1, data.quantity, data.items, data.total));
-      });
+    // Only make a request if the user has been set.
+    if (isUserReady()) {
+      fetch(userFilter);
     }
-  }, [api, users]);
+    // Only want to make a request when filter or sort change.
+    // 'fetch' regrettably changes any time the advanced filter changes.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [userFilter, isUserReady()]);
 
   return (
     <styled.UserList>
       <FormPage>
         <PagedTable
           columns={columns}
           header={UserFilter}
+          sorting={{ sortBy: userFilter.sort }}
           isLoading={!!requests.length}
           page={page}
           onRowClick={(row) => navigate(`${row.original.id}`)}
+          onChangeSort={handleChangeSort}
```

Comment:
Did you find that the `onChangeSort` gets called a lot when testing?  I was surprised how often react.table detects a change.

Reply:
Yes, couldn't figure out all the triggers...